R/add_CI.R

```diff
@@ -37,31 +37,53 @@ add_CI <- function(gg, ...){
 #' @method add_CI ggsurvfit
 #' @export
 
-add_CI.ggsurvfit <- function(gg, alpha = 0.1, style = "ribbon", linetype = 2, ...){
-
+add_CI.ggsurvfit <- function(gg, 
+                             alpha = 0.1, 
+                             style = "ribbon", 
+                             linetype = "2a", ...){
+  
   if (! base::all(c("est.lower", "est.upper") %in% colnames(gg$data))) {
     warning("Confidence limits were not part of original estimation.")
     return(NULL)
   }
-
+  
   if (! base::any(c("ribbon", "step") %in% style)) {
     warning("Invalid `step` argument.")
     return(NULL)
   }
-
+  
   if (style == "ribbon"){
-    warning("Argument linetype not used for style ribbon")
     gg <- gg +
-      ggplot2::geom_ribbon(ggplot2::aes(ymin = est.lower, ymax = est.upper, fill = strata), alpha = alpha, na.rm = TRUE)
+      ggplot2::geom_ribbon(ggplot2::aes(ymin = est.lower, 
+                                        ymax = est.upper, 
+                                        fill = strata), 
+                           alpha = alpha, 
+                           na.rm = TRUE)
   }
-
+  
   if (style == "step"){
+    
+    if (linetype == "2a") {
```

Comment:
I dont understand the issue here.

Reply:
There isn't really an "issue", but calling the function with its default parameters yields a warning. Rewriting it like this doesn't change the behavior as presented to the user but doesn't cause a warning. In my opinion, the default behavior of shouldn't automatically cause a warning since that'd motivate (at least me) to investigate this further.

Reply:
![image](https://user-images.githubusercontent.com/12818242/110389064-45bc8600-8064-11eb-8093-7031af4228eb.png)
This is the current behavior

Reply:
Without giving any further parameters, it defaults to `style = "ribbon"` and `linetype = 2`. Right now, this triggers the warning in line 53 (`warning("Argument linetype not used for style ribbon")`) which the user per se did not cause. This would irritate me personally. I've now pushed a less hacky looking fix that now only prints a warning if the user manually specified a linetype while using ribbon and sets it to solid when using step.